src/v/kafka/group_probe.h

```diff
@@ -46,9 +54,87 @@ class group_offset_probe {
             labels)});
     }
 
+    void setup_public_metrics(
+      const kafka::group_id& group_id, const model::topic_partition& tp) {
+        namespace sm = ss::metrics;
+
+        if (config::shard_local_cfg().disable_public_metrics()) {
+            return;
+        }
+
+        auto group_label = sm::label("group");
+        auto topic_label = sm::label("topic");
+        auto partition_label = sm::label("partition");
+        std::vector<sm::label_instance> labels{
+          group_label(group_id()),
+          topic_label(tp.topic()),
+          partition_label(tp.partition())};
+
+        _public_metrics.add_group(
+          prometheus_sanitize::metrics_name("kafka:consumer:group"),
+          {sm::make_gauge(
+             "committed_offset",
+             [this] { return _offset; },
+             sm::description("Consumer group committed offset"),
+             labels)
+             .aggregate({sm::shard_label})});
+    }
+
 private:
     model::offset& _offset;
     ss::metrics::metric_groups _metrics;
+    ss::metrics::metric_groups _public_metrics;
+};
+
+template<typename KeyType, typename ValType>
+class group_probe {
+    using member_map = absl::node_hash_map<kafka::member_id, member_ptr>;
+    using static_member_map
+      = absl::node_hash_map<kafka::group_instance_id, kafka::member_id>;
+
+public:
+    explicit group_probe(
+      member_map& members,
+      static_member_map& static_members,
+      absl::node_hash_map<KeyType, ValType>& offsets) noexcept
```

Comment:
nit: This could have a type alias like the others.

Reply:
Done.